[DataGrid] New Toolbar component (#14611)
mui/mui-x: docs/data/data-grid/components/export/export.md

# Data Grid - Export component

<p class="description">Components to trigger exports of the Data Grid.</p>

## Anatomy

The Export component is comprised of the following parts:

```tsx
<Grid.Export.Trigger />
```

### Trigger

A button that triggers an export.

The `exportType` prop can be set to `print`, `csv` or `excel`[<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan').

## Examples

Below are some ways the Export component can be used.

### Toolbar export buttons

Display export options as buttons on the toolbar.

{{"demo": "GridExportTrigger.js", "bg": "inline"}}

### Toolbar export menu

Display export options within a menu on the toolbar.

{{"demo": "GridExportMenu.js", "bg": "inline"}}

## API

> - TBC

Author: Having issues generating the API page for this currently, as there is a `GridExportTrigger` exported from both the community and premium package. I think having a single component which can take different `exportType` values depending on the license is the best DX here, so keen not to lose that.

Reviewer: Side note: I noticed that `GridExportTrigger` in the premium package reimplements the one from the community package + adds Excel export. So the component from the premium package won't inherit the changes made to the community one, and it's easy to miss.

Did you consider having separate components for each export type? In this case, we'll export an additional trigger component for Excel export. What do you think?

Author: > So the component from the premium package won't inherit the changes made to the community one, and it's easy to miss.

Fair point - I've split it out into three separate components to avoid this.

---
title: React Data Grid - Export component
productId: x-data-grid
components: ExportPrint, ExportCsv, ExportExcel
packageName: '@mui/x-data-grid'
githubLabel: 'component: data grid'
---

# Data Grid - Export

> <p class="description">Components to trigger exports of the Data Grid.</p>

Reviewer 2: to make it a little more clear what's meant by "export" in this context

```suggestion
<p class="description">Let users export the Data Grid for Excel, CSV, or printing.</p>
```

## Basic usage

The demo below shows how to add export triggers to a custom toolbar.

{{"demo": "GridExportTrigger.js", "bg": "inline", "defaultCodeOpen": false}}

## Anatomy

```tsx
import { ExportPrint, ExportCsv, ExportExcel } from '@mui/x-data-grid';

<ExportPrint />
<ExportCsv />
<ExportExcel />
```

> ### ExportPrint

Reviewer 2: same as the other doc - plain text is preferred for headers whenever possible

```suggestion
### Export Print
```

---
title: React Data Grid - Export component
productId: x-data-grid
components: ExportPrint, ExportCsv, ExportExcel
packageName: '@mui/x-data-grid'
githubLabel: 'component: data grid'
---

# Data Grid - Export

<p class="description">Components to trigger exports of the Data Grid.</p>

## Basic usage

The demo below shows how to add export triggers to a custom toolbar.

{{"demo": "GridExportTrigger.js", "bg": "inline", "defaultCodeOpen": false}}

## Anatomy

```tsx
import { Export } from '@mui/x-data-grid';

<Export.Print />
<Export.Csv />
<Export.Excel />
```

> ### Print

Author: Replied here https://github.com/mui/mui-x/pull/14611#discussion_r1918253099

A button that triggers a print export.

Renders the `baseButton` slot.

### Csv

A button that triggers a CSV export.

Renders the `baseButton` slot.

### Excel [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

A button that triggers an Excel export.

Renders the `baseButton` slot.

## Recipes

Below are some ways the Export component can be used.

> 

Reviewer 3: This feels unnecessary given that there's only one demo. I would cut it and just go straight from `## Excel` to `### Toolbar export menu`

Author: > This feels unnecessary given that there's only one demo

Yeah, fair point. I was trying to create a structure that works across all of the component pages, some of which will have multiple recipes. Probably not necessary though.

Reviewer 3: Ah I see, if it's helpful on other pages then I'm all for it—I'd rather see consistency across similar pages, so the reader knows exactly what to expect as they move between them.

### Toolbar export menu

> Display export options within a menu on the toolbar.

Reviewer 3: ```suggestion
The demo below shows how to display export options within a menu on the toolbar.
```

{{"demo": "GridExportMenu.js", "bg": "inline", "defaultCodeOpen": false}}

> ## Custom elements

Reviewer 3: same feedback as Columns Panel page

The default elements can be replaced using the `render` prop. See [Grid components—Customization](/x/react-data-grid/components/overview/#customization).

See the [Toolbar—Custom elements demo](/x/react-data-grid/components/toolbar/#custom-elements) for an example.

## Accessibility

### ARIA

> - The element rendered by the `<Export.Print />`, `<Export.Csv />` and `<Export.Excel />` components should have a text label, or an `aria-label` attribute set.

Reviewer 3: same feedback as Columns Panel
